### ISSUE TITLE
BUG: np.random.Generator.multinomial example with multiple sets of p values doesn't work

### ISSUE THREAD

**Issue by rishi-kulkarni** [numpy/numpy#21229]
### Describe the issue:

The random.Generator.multinomial documentation includes an example where the multinomial function is given multiple sets of p-values. However, this appears to throw an error.

### Reproduce the code example:

```python
rng = np.random.default_rng()
rng.multinomial([10, 20],[[1/4]*4 + [0]*2, [1/6]*6])
```

### Error message:

```shell
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [15], in <cell line: 2>()
      1 rng = np.random.default_rng()
----> 2 rng.multinomial([10, 20],[[1/4]*4 + [0]*2, [1/6]*6])

File _generator.pyx:3757, in numpy.random._generator.Generator.multinomial()

ValueError: object too deep for desired array
```

### NumPy/Python version information:

1.21.5

**Comment by charris** [comment]
Works for me in main, it seems to have been fixed.
```
In [1]: rng = np.random.default_rng()
   ...: rng.multinomial([10, 20],[[1/4]*4 + [0]*2, [1/6]*6])
Out[1]: 
array([[4, 3, 1, 2, 0, 0],
       [6, 2, 4, 1, 1, 6]])

In [2]: np.__version__
Out[2]: '1.23.0.dev0+815.ga804f89415'
```

**Comment by rishi-kulkarni** [comment]
Yes, it looks like it's fixed as early as 1.22.3. Closing this for now.
